lib/cfn-model/parser/api_gateway_stage_parser.rb

```diff
@@ -9,21 +9,21 @@ class ApiGatewayStageParser
   def parse(cfn_model:, resource:)
     api_stage = resource
 
-    attach_usage_plans_to_api_stage(cfn_model: cfn_model, api_stage: api_stage)
+    attach_usage_plan_to_api_stage(cfn_model: cfn_model, api_stage: api_stage)
     attach_deployment_id_to_api_stage(cfn_model: cfn_model, api_stage: api_stage)
     api_stage
   end
 
   private
 
-  def attach_usage_plans_to_api_stage(cfn_model:, api_stage:)
+  def attach_usage_plan_to_api_stage(cfn_model:, api_stage:)
     usage_plans = cfn_model.resources_by_type 'AWS::ApiGateway::UsagePlan'
-    usage_plans.select do |usage_plan|
+    usage_plans.each do |usage_plan|
       next if usage_plan.apiStages.nil?
-      usage_plan.apiStages.select do |up_api_stage|
+      usage_plan.apiStages.each do |up_api_stage|
         if up_api_stage['Stage'].is_a?(Hash) && up_api_stage['Stage'].key?('Ref')
```

Comment:
anything in References that can hide this?  @pshelby recently added some methods there

Reply:
@tmcelhattan [`References.resolve_resource_id()`](https://github.com/stelligent/cfn-model/blob/0a590d92131e96f53097abd9b4942f249337f41f/lib/cfn-model/model/references.rb#L32-L46) might be able to help here.  Lmk if you need more info.  [Example usage](https://github.com/stelligent/cfn-model/blob/0a590d92131e96f53097abd9b4942f249337f41f/lib/cfn-model/model/cfn_model.rb#L82)